pkg/reconciler/route/resources/certificate.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/knative/pkg/system"
+	networkingv1alpha1 "github.com/knative/serving/pkg/apis/networking/v1alpha1"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/pkg/reconciler/route/resources/names"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// MakeCertificates creates Certificates for the Route to request TLS certificates.
+func MakeCertificates(route *v1alpha1.Route, dnsNames []string, enableWildcardCert bool) []*networkingv1alpha1.Certificate {
```

Comment:
High-level concern: I don't think creation of the wildcard certificates belongs here.

I think we should have a Namespace controller that eagerly creates a wildcard certificate per Namespace.

I think the logic here should check for a wildcard, but fallback on non-wildcard synthesis (with clear ownership of the lifecycle of those Certificates).

I think to unblock this PR, let's focus on the non-wildcard case?  Happy to discuss.

Reply:
Per our discussion, we will focus on non-wildcard in this PR, and work on wildcard scenario later.